nemo/collections/nlp/parts/nlp_overrides.py

```diff
@@ -451,6 +468,299 @@ def _configure_launcher(self):
+            )
+
+        if not HAVE_MEGATRON_CORE:
+            raise ImportError(
+                "megatron-core was not found. Please see the NeMo README for installation instructions: https://github.com/NVIDIA/NeMo#megatron-gpt."
+            )
+
+        # Set the mixed precision recipe
+        kwargs['mixed_precision'] = self._set_mixed_precision_recipe(precision, grad_reduce_dtype)
+        # Use the default FSDP backward-prefetch policy for proper communication overlap.
+        kwargs['backward_prefetch'] = BackwardPrefetch.BACKWARD_PRE
+
+        # Set FSDP wrapping policy: use Transformer layer module as the FSDP sharding granularity.
+        self.fsdp_wrap_module = {
+            MCoreTransformerLayer,
+            AutocastTransformerLayer,
+            ParallelTransformerLayer,
+        }
+        kwargs['auto_wrap_policy'] = functools.partial(
+            transformer_auto_wrap_policy, transformer_layer_cls=self.fsdp_wrap_module
+        )
+
+        # Set FSDP sharding strategy.
+        fsdp_sharding_strategy = {
+            'full': ShardingStrategy.FULL_SHARD,
+            'hybrid': ShardingStrategy.HYBRID_SHARD,
+            'grad': ShardingStrategy.SHARD_GRAD_OP,
+        }
+        assert sharding_strategy in list(fsdp_sharding_strategy.keys()), "Not a supported sharding strategy."
+        assert sharding_strategy != 'hybrid', "Hybrid sharding is currrently not supported."
+        kwargs['sharding_strategy'] = fsdp_sharding_strategy[sharding_strategy]
+
+        # Set FSDP state dict configs
+        self.sharded_checkpoint = sharded_checkpoint
+        self.state_dict_context = (
+            _get_sharded_state_dict_context if sharded_checkpoint else _get_full_state_dict_context
+        )
+
+        self.nccl_communicator_config_path = nccl_communicator_config_path
+        super().__init__(**kwargs)
+
+    def _set_mixed_precision_recipe(
+        self, precision: Union[int, str], grad_reduce_dtype: Union[int, str]
+    ) -> MixedPrecision:
+        """
+        Set FSDP mixed precision recipe.
+        `param_dtype` sets the data type for computation in forward and backpropagation, and the parameter
+        data type for optimizer execution is maintained in the full precision.
+        `buffer_dtype` is only valid when a module has buffers by `register_buffer` method, which is not
+        shared by FSDP.
+        `reduce_dtype` sets gradient reduction data type.
+        """
+        if precision in ["16-true", "16-mixed", 16]:
+            param_dtype = reduce_dtype = buffer_dtype = torch.float16
+        elif precision in ["bf16-true", "bf16-mixed", "bf16"]:
+            param_dtype = reduce_dtype = buffer_dtype = torch.bfloat16
+        elif precision == 32:
+            param_dtype = reduce_dtype = buffer_dtype = torch.float
+        else:
+            raise ValueError(f"Was unable to infer precision type, received {precision!r}.")
```

Comment:
Again for a future PR but we need to use a utility function for this that can be shared across all of nemo.